src/Nethermind/Nethermind.Merge.Plugin/Handlers/ForkchoiceUpdatedHandler.cs

```diff
@@ -464,4 +439,52 @@ private BlockInfo[] GetBeaconChainBranch(Block newHeadBlock, BlockInfo newHeadBl
 
         return blocksList.ToArray();
     }
+
+    private class StateRootUpdater : IDisposable
+    {
+        private readonly IWorldState _globalWorldState;
+        private readonly IBlockProcessingQueue _queue;
+        private Hash256? _stateRoot;
+
+        public StateRootUpdater(IWorldState globalWorldState, IBlockProcessingQueue queue)
+        {
+            _globalWorldState = globalWorldState;
+            _queue = queue;
+            _queue.ProcessingQueueEmpty += OnProcessingQueueEmpty;
+        }
+
+        public Hash256? StateRoot
```

Comment:
So _globalWorldState.StateRoot can be potentially directly changed from other place of the app. Seems a bit unsafe. Maybe we can have `_globalWorldState.EnqueueStateSelection(newStateRoot)` and read only `_globalWorldState.StateRoot`?

Reply:
It is under lock in general and state root shouldn't be changed from anywhere eslse.